test/unit-tests/sourcekit-lsp/LanguageClientManager.test.ts

```diff
@@ -0,0 +1,408 @@
+            when(languageClientMock.state).thenReturn(languageClient.State.Running);
+            lspConfig.setValue({
+                disable: true,
+                serverPath: "/path/to/sourcekit-lsp",
+                serverArguments: [],
+                inlayHintsEnabled: false,
+                supportCFamily: "disable",
+                supportedLanguages: [],
+            });
+
+            onDidChangeConfigurationMock.notifyAll(instance(eventMock));
+
+            verify(windowMock.showInformationMessage(anyString(), "Stop Language Server")).once();
+        });
+
+        test("Already stopped", async () => {
+            when(languageClientMock.state).thenReturn(languageClient.State.Stopped);
+            lspConfig.setValue({
+                disable: true,
+                serverPath: "/path/to/sourcekit-lsp",
+                serverArguments: [],
+                inlayHintsEnabled: false,
+                supportCFamily: "disable",
+                supportedLanguages: [],
+            });
+
+            onDidChangeConfigurationMock.notifyAll(instance(eventMock));
+
+            verify(windowMock.showInformationMessage(anyString(), anyString())).never();
+        });
+
+        test("Start", async () => {
+            when(languageClientMock.state).thenReturn(languageClient.State.Stopped);
+            lspConfig.setValue({
+                disable: false,
+                serverPath: "/path/to/sourcekit-lsp",
+                serverArguments: [],
+                inlayHintsEnabled: false,
+                supportCFamily: "disable",
+                supportedLanguages: [],
+            });
+
+            onDidChangeConfigurationMock.notifyAll(instance(eventMock));
+
+            verify(windowMock.showInformationMessage(anyString(), "Start Language Server")).once();
+        });
+
+        test("Restart", async () => {
+            when(languageClientMock.state).thenReturn(languageClient.State.Running);
+            when(eventMock.affectsConfiguration("swift.sourcekit-lsp.disable")).thenReturn(false);
+            lspConfig.setValue({
+                disable: false,
+                serverPath: "/path/to/other/sourcekit-lsp",
+                serverArguments: [],
+                inlayHintsEnabled: false,
+                supportCFamily: "disable",
+                supportedLanguages: [],
+            });
+
+            onDidChangeConfigurationMock.notifyAll(instance(eventMock));
+
+            verify(
+                windowMock.showInformationMessage(anyString(), "Restart Language Server")
+            ).once();
+        });
+    });
+
+    suite("useLanguageClient", () => {
+        test("Server startup succeeds", async () => {
+            sut = new LanguageClientManager(instance(workspace));
+
+            await sut.useLanguageClient(
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                async (client, token) => {
+                    assert.notEqual(client, undefined);
```

Comment:
nit: I prefer `assert.ok(client)` to check for truthyness

Reply:
done